x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/crud/rules_management_client.ts

```diff
@@ -0,0 +1,379 @@
+): Promise<RuleAlertType> => {
+  const { ruleAsset } = createPrebuiltRulePayload;
+
+  const rule = await _createRule(rulesClient, ruleAsset, {
+    immutable: true,
+    defaultEnabled: false,
+  });
+
+  return rule;
+};
+
+export const updateRule = async (
+  rulesClient: RulesClient,
+  updateRulePayload: UpdateRuleProps
+): Promise<RuleAlertType> => {
+  const { ruleUpdate, existingRule } = updateRulePayload;
+
+  const update = await _updateRule(rulesClient, { ruleUpdate, existingRule });
+
+  const enabled = ruleUpdate.enabled ?? existingRule.enabled;
+  await _toggleRuleEnabledOnUpdate(rulesClient, existingRule, enabled);
+
+  return { ...update, enabled };
+};
+
+export const patchRule = async (
+  rulesClient: RulesClient,
+  patchRulePayload: PatchRuleProps
+): Promise<RuleAlertType> => {
+  const { nextParams, existingRule } = patchRulePayload;
+  const update = await _patchRule(rulesClient, patchRulePayload);
+
+  await _toggleRuleEnabledOnUpdate(rulesClient, existingRule, nextParams.enabled ?? false);
+
+  if (nextParams.enabled != null) {
+    return { ...update, enabled: nextParams.enabled };
+  } else {
+    return update;
+  }
+};
+
+export const deleteRule = async (
+  rulesClient: RulesClient,
+  deleteRulePayload: DeleteRuleProps
+): Promise<void> => {
+  const { ruleId } = deleteRulePayload;
+  await rulesClient.delete({ id: ruleId });
+};
+
+export const upgradePrebuiltRule = async (
+  rulesClient: RulesClient,
+  upgradePrebuiltRulePayload: UpgradePrebuiltRuleProps
+): Promise<RuleAlertType> => {
+  const { ruleAsset } = upgradePrebuiltRulePayload;
+  const existingRule = await readRules({
+    rulesClient,
+    ruleId: ruleAsset.rule_id,
+    id: undefined,
+  });
+
+  if (!existingRule) {
+    throw new PrepackagedRulesError(`Failed to find rule ${ruleAsset.rule_id}`, 500);
+  }
+
+  // If rule has change its type during upgrade, delete and recreate it
+  if (ruleAsset.type !== existingRule.params.type) {
+    return _upgradePrebuiltRuleWithTypeChange(rulesClient, ruleAsset, existingRule);
+  }
+
+  // Else, simply patch it.
+  await _patchRule(rulesClient, { existingRule, nextParams: ruleAsset });
+
+  const updatedRule = await readRules({
+    rulesClient,
+    ruleId: ruleAsset.rule_id,
+    id: undefined,
+  });
+
+  if (!updatedRule) {
+    throw new PrepackagedRulesError(`Rule ${ruleAsset.rule_id} not found after upgrade`, 500);
+  }
+
+  return updatedRule;
+};
+
+export const importRule = async (
+  rulesClient: RulesClient,
+  importRulePayload: ImportRuleProps
+): Promise<RuleAlertType> => {
+  const { ruleToImport, overwriteRules, options } = importRulePayload;
+
+  const existingRule = await readRules({
+    rulesClient,
+    ruleId: ruleToImport.rule_id,
+    id: undefined,
+  });
+
+  if (!existingRule) {
+    return _importNewRule(rulesClient, { ruleToImport, options });
+  } else if (existingRule && overwriteRules) {
+    return _importExistingRule(rulesClient, { ruleToImport, existingRule });
```

Comment:
Same with `_importNewRule` and `_importExistingRule`, let's inline them to improve readability